src/NuGet.Core/NuGet.ProjectModel/PackageSpecOperations.cs

```diff
@@ -13,8 +13,11 @@ namespace NuGet.ProjectModel
 {
     public static class PackageSpecOperations
     {
-        public static void AddOrUpdateDependency(PackageSpec spec, PackageDependency dependency)
```

Comment:
we are removing a public api that has potential to break sdk/cli when they update.
i would prefer if you added an overload and make this method call the overloaded method.